immuno_graphical/verification/verification_report.Rmd

```diff
@@ -0,0 +1,463 @@
+    
+    if(is.list(round_digits)){
+      if(column_name %in% names(round_digits)){
+        round_digit <- round_digits[[column_name]]
+      }else{
+        round_digit <- round_digits[[".default"]]
+      }
+    }else{
+      round_digit <- round_digits
+    }
+    
+    if(is.list(thresholds)){
+      if(column_name %in% names(thresholds)){
+        threshold <- thresholds[[column_name]]
+      }else{
+        threshold <- thresholds[[".default"]]
+      }
+    }else{
+      threshold <- thresholds
+    }
+    
+    
+    
+    v1 <- naturalize( ds1[ order( ds1[[index]] ), column_name, drop = TRUE], round_digits = round_digit)
+    v2 <- naturalize( ds2[ order( ds2[[index]] ), column_name, drop = TRUE], round_digits = round_digit)
+    is_equal <- equal( v1, v2 , threshold = threshold)
+    if ( !all( is_equal ) & verbose)
+      print( paste0("`", column_name, "` is not equal. ", sum(!is_equal),"/", max(length(v1),length(v2)), " mismatches." ) )
+    
+    
+    
+    return( list(
+      equal = all( is_equal ),
+      diffs = data.frame(
+        key = sort(ds1[[index]])[!is_equal],
+        ds1 = v1[!is_equal],
+        ds2 = v2[!is_equal],
+        stringsAsFactors = FALSE
+        ))
+      )
+  })
+  
+  names(res) <- cols
+  
+  resmatched <- do.call( 'c', lapply(res,`[[`,1))
+  if( verbose ){
+    message( "There are ", sum(!resmatched), " mismatched fields of ", length(resmatched),'.')
+  }
+  
+  attr(res,"index") <- index
+  
+  class(res) <- "comparison"
+  return( res )
+}
+print.comparison <- function(x,...){
+  n_mismatch <- do.call('c',lapply(x,function(y){
+    z <- nrow(y$diffs)
+    if(z>0){
+      z
+    }else{
+      NULL
+    }
+  }))
+  
+  for(comp in names(n_mismatch)){
+      message( paste0("`", comp, "` is not equal. ", n_mismatch[[comp]], " mismatches." ) )
+  }
+}
+rerun_failed_comparisons <- function( x, ds1, ds2){
+  mismatched <- names(do.call('c',lapply(x,function(y){
+    if(nrow(y$diffs)>0){
+      1
+    }else{
+      NULL
+    }
+  })))
+  compare_datasets(mismatched, ds1, ds2, index = attr(x,"index"))
+}
+reformat <- function(x, digits = 1){
+  format(round(as.numeric(x),digits = digits), nsmall = digits)
+}
+```
+
+## Description
+
+This is the verification report for the `immuno_graphical` folder of the `correlates_reporting` project for CoVPN. 
+
+In the verification process, the outputs of `immuno_graphical_verification_code.R` were visually compared against the plots listed in the file `immuno_graphical/verification/verification_plan.pdf`. The values presented in the original plots were also generated independently by the tester and compared against the outputs generated by the original programmer.
+
+The files `immuno_graphical/verification/verification_input/practice_data.csv`, `immuno_graphical/verification/twophase_data.csv` and 
+`immuno_graphical/verification/long_twophase_data.csv` were provided by the original programmer to the tester for verification purposes.
```

Comment:
I see you are reading/loading additional files. Could you list them as well. Also, check the PDF, the text is floating off the edge for some lines. insert a `\newline` into the text to get it to wrap down.

Reply:
Updated.